### ISSUE TITLE
RecyclerViewDragDropManager - Interception of action up if long pressed weren't triggered

### ISSUE THREAD

**Issue by MTomczynski** [h6ah4i/android-advancedrecyclerview#310]
Hello,
I was wondering if is there any simple way to intercept touch event on an item when long press is set up but not fired by the touch.

What I need to do is to set action up with my custom code, and drag and drop operation on long press. Just as in Your example draggable grid but with custom code after long press fails. Listeners won't work inside adapter, not sure why, maybe because wrapping in DragDropManager?

**Comment by h6ah4i** [comment]
@MTomczynski Hi. Thank you for using my library :smile:

### TL;DR

Set `OnClickListener` to the `mContainer` instead of the `itemView`.

```java
public void onBindViewHolder(MyViewHolder holder, final int position) {
    holder.mContainer.setOnClickListener(...);
}
```

### Explanations

The `mContainer` (id: `@+id/container`) view has an `android:clickable="true"` attribute, so this view consumes click events. The clickable attribute is required to make foreground drawable respond to touch events.

**Layout XML:**

```xml
<!-- itemView -->
<FrameLayout
    style="@style/commonListItemStyle"
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="96dp"
    android:layout_margin="4dp"
    android:background="@drawable/bg_swipe_item_neutral">

    <!-- mContainer -->
    <FrameLayout
        android:id="@+id/container"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:clickable="true"
        android:foreground="?attr/selectableItemBackground"
        tools:ignore="UselessParent">
        <TextView
            android:id="@android:id/text1"
            android:layout_width="match_parent"
            android:layout_height="match_parent"
            android:layout_gravity="top|left"
            android:gravity="center"
            tools:ignore="RtlHardcoded"/>
    </FrameLayout>
</FrameLayout>
```

Note that the container view is only required when using swiping feature. If you want to use draggable feature only, it can simply be removed.

---

ref.)  Original question on StackOverflow:

- [android - Advanced Recycler View - Handling on item click and long click drag operations - Stack Overflow](http://stackoverflow.com/questions/39828088/advanced-recycler-view-handling-on-item-click-and-long-click-drag-operations)

**Comment by MTomczynski** [comment]
Thank You for the response, the problem was that I didn't fully understood your code ;)
